client/elements/text/sc-text-image.html

```diff
@@ -83,6 +91,15 @@
           }
 
           _handleResponse() {
+              if (!this.imageData || !this.imageData[0]) {
+                return;
+              };
+              let pageNumber = 0;
+              for (let item = 0; item < this.imageData.length; item++) {
+                  if (this.imageData[item].pageNumber === this.pageNumber) {
```

Comment:
1. When `this.imageData[item].pageNumber` equals `1`, `item` equals 0. This means `pageNumber` will be always set to `this.imageData[item].pageNumber` minus 1. Is this what we want?
2. `item` refers to an numerical item index, so it's wrongly named.
3. This loop can be simplified to:
```
if (this.imageData.some(page => page.pageNumber === this.pageNumber)) {
    pageNumber = this.pageNumber - 1;
}
```

`some` returns true if there exist any items that satisfy a condition given in the function body.

Reply:
1. Yes, because the array starts at 0 and not at 1 while we start counting at 1.